server/src/main/java/org/spine3/server/command/CommandBus.java

```diff
@@ -185,49 +188,38 @@ public void post(Command command, StreamObserver<Response> responseObserver) {
         }
     }
 
+    /**
+     * Returns {@code true} if a command is valid, {@code false} otherwise.
+     */
     private boolean handleValidation(Command command, StreamObserver<Response> responseObserver) {
+        final Namespace namespace = command.getContext().getNamespace();
+        if (isMultitenant() && isDefault(namespace)) {
+            final Exception exception = InvalidCommandException.onMissingNamespace(command);
+            commandStore.store(command, exception);
+            responseObserver.onError(newInvalidArgumentException(exception));
+            return false; // and nothing else matters
+        }
         final List<ConstraintViolation> violations = CommandValidator.getInstance().validate(command);
         if (!violations.isEmpty()) {
-            responseObserver.onError(
-                    Status.INVALID_ARGUMENT
-                            .withCause(InvalidCommandException.onConstraintViolations(command, violations))
-                            .asRuntimeException()
-            );
-            return false;
-        }
-        final CommandContext context = command.getContext();
-        if (isMultitenant() && !context.hasNamespace()) {
-            responseObserver.onError(
-                    Status.INVALID_ARGUMENT
-                            .withCause(InvalidCommandException.onMissingNamespace(command))
-                            .asRuntimeException()
-            );
+            final Exception exception = InvalidCommandException.onConstraintViolations(command, violations);
+            commandStore.store(command, exception);
+            responseObserver.onError(newInvalidArgumentException(exception));
             return false;
         }
         return true;
     }
 
     private void handleUnsupported(Command command, StreamObserver<Response> responseObserver) {
         final UnsupportedCommandException exception = new UnsupportedCommandException(command);
-        storeWithErrorStatus(command, exception);
-        responseObserver.onError(
-                Status.INVALID_ARGUMENT
-                        .withCause(exception)
-                        .asRuntimeException()
-        );
+        commandStore.store(command, exception);
+        responseObserver.onError(newInvalidArgumentException(exception));
     }
 
-    /**
-     * Stores a command with the error status.
-     *
-     * @param command a command to store
-     * @param exception an exception occurred during command processing
-     */
-    @VisibleForTesting
-    /* package */ void storeWithErrorStatus(Command command, Exception exception) {
-        store(command);
-        //TODO:2016-05-08:alexander.yevsyukov: Support storage method that can make storing command with error in one call.
-        commandStatusService.setToError(command.getContext().getCommandId(), exception);
+    private static StatusRuntimeException newInvalidArgumentException(Exception exception) {
```

Comment:
The method call sounds a bit misleading. We create new `StatusRuntimeException`, not `IAE`.